cypress/integration/settings/username-change.js

```diff
@@ -0,0 +1,180 @@
+/* global cy */
+
+describe('Username input field', () => {
+  beforeEach(() => {
+    cy.visit('/');
+    cy.contains("Get started (it's free)").click({ force: true });
+    cy.visit('/settings');
+
+    // Setting aliases here
+    cy.get('input[name=username-settings]').as('usernameInput');
+    cy.get('form#usernameSettings').as('usernameForm');
+  });
+
+  it('Should be able to type', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true })
+      .should('have.attr', 'value', 'twaha');
+  });
+
+  it('Should show message when validating name', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true });
+
+    cy.contains('Validating username...')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-info');
+  });
+
+  it('Should show username is avalable if it is avalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('brad', { force: true });
+
+    cy.contains('Username is available.')
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-success');
+  });
+
+  it('Should info message if username is avalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('mruggesh', { force: true });
+
+    cy.contains(
+      'Please note, changing your username will also change ' +
+        'the URL to your profile and your certifications.'
+    )
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-info');
+  });
+
+  // eslint-disable-next-line
+  it('Should be able to click the `Save` button if username is avalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('ojeyton', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('not.be.disabled');
+    });
+  });
+
+  it('Should show username is unavalable if it is unavalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true });
+
+    cy.contains('Username not available.')
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-warning');
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if username is unavalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
```

Comment:
```suggestion
    cy.get('@usernameForm')
      .contains('Save')
      .should('be.disabled');
```
`contains` acts on the result of `get` so this can just be chained.